sysl2/sysl/cmd_export.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	sysl "github.com/anz-bank/sysl/src/proto"
+	"github.com/anz-bank/sysl/sysl2/sysl/exporter"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type exportCmd struct {
+	appName string
+	out     string
+	mode    string
+	format  string
+}
+
+const (
+	swaggerMode = "swagger"
+	jsonMode    = "json"
+	yamlMode    = "yaml"
+)
+
+func (p *exportCmd) Name() string            { return "export" }
+func (p *exportCmd) RequireSyslModule() bool { return true }
+
+func (p *exportCmd) Configure(app *kingpin.Application) *kingpin.CmdClause {
+	cmd := app.Command(p.Name(), "Export sysl to external types. Supported types: Swagger")
+	cmd.Flag("app-name",
+		"name of the sysl app defined in sysl model."+
+			" if there are multiple apps defined in sysl model,"+
+			" swagger will be generated only for the given app").Default("all").Short('a').StringVar(&p.appName)
+	cmd.Flag("output",
```

Comment:
I agree that the design is minimalistic but it works well. I agree that the user may not look at sysl --help but the hintoptions and error messages can provide the necessary info.

Reply:
It is very important to try to get user-facing changes as close to perfect as can be from the beginning, otherwise it adds a huge barrier to change later on. Even if the only people using the app are ourselves, and even if it is feature we may not use much, it is still important to get it right first go.

This is 100x more important with adding new API's because you never want to break other peoples code (the linux kernel has an almost concrete rule that no matter how big a bug or feature is, if it breaks external API or ABI it will be rejected).

Reply:
Design doesn't have to elaborate to work, it is important to get the things right I agree but that doesn't mean you should choose an overtly complex design each time.